### ISSUE TITLE
[java] False positive for rule UselessParentheses

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2910]
**Affects PMD Version:** 
6.29.0

**Description:**

False positive for rule [UselessParentheses](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#uselessparentheses) when concatenating string constant and string variable within parentheses, subsequently calling method on concatenated result. The parentheses are incorrectly flagged.

**Code Sample demonstrating the issue:**

```java
package jls;

public final class TestStringInterned {
  
  public static void main(String[] args) {
    String hello = "Hello";
    String lo = "lo";
System.out.println(hello == ("Hel" + lo).intern());
    }
}
```

**Steps to reproduce:**

Execute PMD on above class for rule UselessParentheses.

**Running PMD through:** 
**[ Ant]**

**Comment by Fernal73** [comment]
Is this a duplicate of [this](https://github.com/pmd/pmd/issues/2739)? Kindly close if it is.

**Comment by oowekyala** [comment]
I think it is, thanks for pointing it out. I added the test case on the PMD7 branch.
